reader-osm/src/main/java/com/graphhopper/reader/osm/OSMReader.java

```diff
@@ -411,14 +414,16 @@ void processWay(ReaderWay way) {
 
     public void processRelation(ReaderRelation relation) {
         if (relation.hasTag("type", "restriction")) {
-            OSMTurnRelation turnRelation = createTurnRelation(relation);
-            if (turnRelation != null) {
-                GraphExtension extendedStorage = graph.getExtension();
-                if (extendedStorage instanceof TurnCostExtension) {
-                    TurnCostExtension tcs = (TurnCostExtension) extendedStorage;
-                    Collection<TurnCostTableEntry> entries = analyzeTurnRelation(turnRelation);
-                    for (TurnCostTableEntry entry : entries) {
-                        tcs.addTurnInfo(entry.edgeFrom, entry.nodeVia, entry.edgeTo, entry.flags);
+            List<OSMTurnRelation> turnRelations = createTurnRelation(relation);
+            for (OSMTurnRelation turnRelation : turnRelations) {
+                if (turnRelation != null) {
+                    GraphExtension extendedStorage = graph.getExtension();
+                    if (extendedStorage instanceof TurnCostExtension) {
```

Comment:
In case we have no turn cost extension we can return early here (we can change the order of the if statement and the for loop)